dask/typing.py

```diff
@@ -0,0 +1,335 @@
+            final key along with optional arguments. An example signature
+            would be ``finalize(results: Sequence[Any], *args)``.
+        tuple[Any, ...]
+            Optional arguments passed to the function following the
+            key results (the `*args` part of the
+            ``PostComputeCallable``. If no additional arguments are to
+            be passed then this must be an empty tuple.
+
+        """
+        raise NotImplementedError("Inheriting class must implement this method.")
+
+    @abc.abstractmethod
+    def __dask_postpersist__(self) -> tuple[PostPersistCallable, tuple]:
+        """Rebuilder function and optional arguments to contruct a persisted collection.
+
+        Returns
+        -------
+        PostPersistCallable
+            Callable that rebuilds the collection. The signature
+            should be
+            ``rebuild(dsk: Mapping, *args: Any, rename: Mapping[str, str] | None)``.
+            The callable should return an equivalent Dask collection
+            with the same keys as `self`, but with results that are
+            computed through a different graph. In the case of
+            :py:func:`dask.persist`, the new graph will have just the
+            output keys and the values already computed.
+        tuple[Any, ...]
+            Optional arugments passed to the rebuild callable. If no
+            additional arguments are to be passed then this must be an
+            empty tuple.
+
+        """
+        raise NotImplementedError("Inheriting class must implement this method.")
+
+    @abc.abstractmethod
+    def __dask_tokenize__(self) -> Hashable:
+        """Value that must fully represent the object."""
+        raise NotImplementedError("Inheriting class must implement this method.")
+
+    __dask_optimize__: Any
+    """Given a graph and keys, return a new optimized graph.
+
+    This method can be either a ``staticmethod`` or a ``classmethod``,
+    but not an ``instancemethod``. For example implementations see the
+    definitions of ``__dask_optimize__`` in the core Dask collections:
+    ``dask.array.Array``, ``dask.dataframe.DataFrame``, etc.
+
+    Note that graphs and keys are merged before calling
+    ``__dask_optimize__``; as such, the graph and keys passed to
+    this method may represent more than one collection sharing the
+    same optimize method.
+
+    Parameters
+    ----------
+    dsk : MutableMapping
+        The merged graphs from all collections sharing the same
+        ``__dask_optimize__`` method.
+    keys : Sequence[Hashable]
+        A list of the outputs from ``__dask_keys__`` from all
+        collections sharing the same ``__dask_optimize__`` method.
+    **kwargs : Any
+        Extra keyword arguments forwarded from the call to
+        ``compute`` or ``persist``. Can be used or ignored as
+        needed.
+
+    Returns
+    -------
+    MutableMapping
+        The optimized Dask graph.
+
+    """
+
+    __dask_scheduler__: staticmethod
```

Comment:
adding the bracketed type didn't seem to make a difference (in mypy or pyright), looks like all the type checkers care about is the `staticmethod` part. I went ahead and added back the `SchedulerGetCallable` since it can serve as at least documentation.